ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java

```diff
@@ -393,7 +393,17 @@ public NumberType getNumberType() throws IOException
                 //Ion decimals can be arbitrary precision, need to read as big decimal
                 return NumberType.BIG_DECIMAL;
             case INT:
-                IntegerSize size = _reader.getIntegerSize();
+                IntegerSize size = null;
+                // Temporary measure until this bug fixing is merged and published
+                // https://github.com/amazon-ion/ion-java/issues/685
+                try {
+                    size = _reader.getIntegerSize();
+                } catch (NullPointerException e) {
+                    // Possible exception
```

Comment:
Hmmh. I assume this is due to invalid/malformer content; and if so, should error out instead of quietly swallowing. Especially since it will fail couple of lines anyway. I can change that.